CMakeLists.txt

```diff
@@ -63,6 +64,7 @@ set_option_if(ACTS_BUILD_EXAMPLES_DD4HEP ACTS_BUILD_EVERYTHING)
 set_option_if(ACTS_BUILD_EXAMPLES_GEANT4 ACTS_BUILD_EVERYTHING)
 set_option_if(ACTS_BUILD_EXAMPLES_HEPMC3 ACTS_BUILD_EVERYTHING)
 set_option_if(ACTS_BUILD_EXAMPLES_PYTHIA8 ACTS_BUILD_EVERYTHING)
+set_option_if(ACTS_BUILD_FATRAS_GEANT4 ACTS_BUILD_EXAMPLES_GEANT4)
```

Comment:
I think there are a few more dependencies that need to be considered:
- If `ACTS_BUILD_FATRAS_GEANT4` is on then `ACTS_BUILD_FATRAS` must be activated as well
- There should also be a dependency on `ACTS_BUILD_EVERYTHING` for completeness.
Since the definition order is important here, this
```
set_option_if(ACTS_BUILD_FATRAS_GEANT4 ACTS_BUILD_EXAMPLES_GEANT4 OR ACTS_BUILD_EVERYTHING)
```
should probably come after the fatras option.

Reply:
Initially `ACTS_BUILD_FATRAS_GEANT4` would have been set if `ACTS_BUILD_EXAMPLES_GEANT4` is set. So the `ACTS_BUILD_EVERYTHING` was covered. At a second glance the G4 flags for Examples and Fatras are independent, so your suggestion should only depend on the second flag. The Fatras flag however should be there, you're right.

Reply:
Yes, the two are independent. I would not rely on interdependencies to satisfy `ACTS_BUILD_EVERYTHING` but to always include it explicitly. The option dependency handling is a bit subtle and this makes it a bit more robust. It is also consistent with how all other options are treated.

Reply:
It is already done